src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Activity.cs

```diff
@@ -0,0 +1,235 @@
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="arguments"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    [FunctionName(nameof(UpdateInstanceWatermarkAsync))]
+    public async Task<IReadOnlyList<InstanceFileIdentifier>> UpdateInstanceWatermarkAsync([ActivityTrigger] BatchUpdateArguments arguments, ILogger logger)
+    {
+        EnsureArg.IsNotNull(arguments, nameof(arguments));
+        EnsureArg.IsNotNull(arguments.InstanceWatermarks, nameof(arguments.InstanceWatermarks));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        logger.LogInformation("Beginning to update all instance watermarks, Total count {TotalCount}", arguments.InstanceWatermarks.Count);
+
+        var instanceWatermarks = arguments.InstanceWatermarks.Select(x => x.Version).ToList();
+
+        IEnumerable<InstanceMetadata> instanceMetadata = await _indexStore.BeginUpdateInstanceAsync(arguments.PartitionKey, instanceWatermarks, CancellationToken.None);
+
+        logger.LogInformation("Completed updating all instance watermarks, Total count {TotalCount}", arguments.InstanceWatermarks.Count);
+
+        return instanceMetadata.Select(x =>
+            new InstanceFileIdentifier
+            {
+                Version = x.VersionedInstanceIdentifier.Version,
+                OriginalVersion = x.InstanceProperties.OriginalVersion,
+                NewVersion = x.InstanceProperties.NewVersion
+            }).ToList();
+    }
+
+    /// <summary>
+    /// Asynchronously batches the instance watermarks and calls the update instance.
+    /// </summary>
+    /// <param name="arguments">BatchUpdateArguments</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the <see cref="UpdateInstanceBatchAsync"/> operation.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="arguments"/> or <paramref name="logger"/> is <see langword="null"/>.
+    /// </exception>
+    [FunctionName(nameof(UpdateInstanceBatchAsync))]
+    public async Task UpdateInstanceBatchAsync([ActivityTrigger] BatchUpdateArguments arguments, ILogger logger)
+    {
+        EnsureArg.IsNotNull(arguments, nameof(arguments));
+        EnsureArg.IsNotNull(arguments.ChangeDataset, nameof(arguments.ChangeDataset));
+        EnsureArg.IsNotNull(arguments.InstanceWatermarks, nameof(arguments.InstanceWatermarks));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        DicomDataset datasetToUpdate = GetDeserialzedDataset(arguments.ChangeDataset);
+
+        int processed = 0;
+
+        logger.LogInformation("Beginning to update all instance blobs, Total count {TotalCount}", arguments.InstanceWatermarks.Count);
+
+        while (processed < arguments.InstanceWatermarks.Count)
+        {
+            int batchSize = Math.Min(_options.BatchSize, arguments.InstanceWatermarks.Count - processed);
```

Comment:
Oh, I see you're using the batch here. This is a bit different from the "fan out" pattern the other functions use -- hence my confusion. The "fan out" refers to the multiple activities that are spawn from a single orchestration to process data in parallel.

I think you can make this sort of batch setting a function-specific option from the config. The reason the other functions get batching options from the input as they have an input on the number of activities that are generated, and as such any change could potentially screw up the orchestration restoration.

Reply:
I will just keep this as a constant in UpdateOptions and not from a host.json.